vision/src/autogluon/vision/detector/detector.py

```diff
@@ -0,0 +1,270 @@
+        config={'log_dir': self._log_dir,
+                'num_trials': 99999 if num_trials is None else max(1, num_trials),
+                'time_limits': 2147483647 if time_limit is None else max(1, time_limit),
+                'search_strategy': search_strategy,
+                'max_reward': max_reward,
+                }
+        if nthreads_per_trial is not None:
+            config['nthreads_per_trial'] = nthreads_per_trial
+        if ngpus_per_trial is not None:
+            config['ngpus_per_trial'] = ngpus_per_trial
+        if isinstance(hyperparameters, dict):
+            if 'batch_size' in hyperparameters:
+                bs = hyperparameters['batch_size']
+                if ngpus_per_trial is not None and ngpus_per_trial > 1 and bs > 16:
+                    # using gpus, check batch size vs. available gpu memory
+                    free_gpu_memory = get_gpu_free_memory()
+                    if not free_gpu_memory:
+                        warnings.warn('Unable to detect free GPU memory, we are unable to verify '
+                                      'whether your data mini-batches will fit on the GPU for the specified batch_size.')
+                    elif len(free_gpu_memory) < ngpus_per_trial:
+                        warnings.warn(f'Detected GPU memory for {len(free_gpu_memory)} gpus but {ngpus_per_trial} is requested.')
+                    elif sum(free_gpu_memory[:ngpus_per_trial]) / bs < 1280:
+                        warnings.warn(f'batch_size: {bs} is potentially larger than what your gpus can support ' +
+                                      f'free memory: {free_gpu_memory[:ngpus_per_trial]} ' +
+                                      'Try reducing "batch_size" if you encounter memory issues.')
+            # check if hyperparameters overwriting existing config
+            for k, v in hyperparameters.items():
+                if k in config:
+                    raise ValueError(f'Overwriting {k} = {config[k]} to {v} by hyperparameters is ambiguous.')
+            config.update(hyperparameters)
+        if scheduler_options is not None:
+            config.update(scheduler_options)
+        # verbosity
+        if log_level > logging.INFO:
+            logging.getLogger('gluoncv.auto.tasks.object_detection').propagate = False
+            for logger_name in ('SSDEstimator', 'CenterNetEstimator', 'YOLOv3Estimator', 'FasterRCNNEstimator'):
+                logging.getLogger(logger_name).setLevel(log_level)
+                logging.getLogger(logger_name).propagate = False
+        task = _ObjectDetection(config=config)
+        task._logger.setLevel(log_level)
+        task._logger.propagate = True
+        self._detector = task.fit(train_data, tuning_data, 1 - holdout_frac, random_state)
+        self._detector._logger.setLevel(log_level)
+        self._detector._logger.propagate = True
+        self._fit_summary = task.fit_summary()
+
+    def _validate_kwargs(self, kwargs):
```

Comment:
consider using `from autogluon.core.scheduler.scheduler_factory import scheduler_factory` to simplify this code and align with Tabular.